src/components/sidebar/Sidebar.js

```diff
@@ -17,23 +17,33 @@ import {
     SCLink,
     Image,
     Sections,
-    SectionsHeader
+    SectionsHeader,
+    ToggleSection,
+    ToggleText
 } from './style';
+import ToggleSwitch from '../shared/toggle-switch/ToggleSwitch';
 import {
     withGameVersionContext,
     withSidebarContext,
     withYokaisContext,
     withBaffleBoardContext,
     withItemsContext,
-    withCrankakaisContext
+    withCrankakaisContext,
+    withThemeContext
 } from '../../store';
 
 class Sidebar extends Component {
     constructor(props) {
         super(props);
+        const { context } = this.props;
+
+        this.state = {
+            toggleEnabled: context.theme
```

Comment:
You did right puting the state globaly in our context, but then here you are replicating the logic again.
For example in `handleChangeTheme()` you do
``` this.setState(state => ({
            toggleEnabled: !state.toggleEnabled
        }));

         context.toggleTheme();
```

So your are changing this to true of false and then at context as well

You could do something like this:

```
      <ToggleSwitch
                                onChange={context.toggleTheme}
                                checked={context.theme}
                            />
```

Reply:
yes you totally rigth, my miss here